src/Optimizer/ImageDimensions.php

```diff
@@ -119,23 +129,46 @@ public function isTiny($threshold = self::TINY_THRESHOLD)
             }
         }
 
+        // If one or both of the dimensions are missing, we cannot deduce an aspect ratio.
         if ($this->getWidth() === null || $this->getHeight() === null) {
             return true;
         }
 
+        // If one or both of the dimensions are zero, the entire image will be invisible.
         if (
             (is_numeric($this->getWidth()) && $this->getWidth() <= 0)
             || (is_numeric($this->getHeight()) && $this->getHeight() <= 0)
         ) {
             return true;
         }
 
+        // If relative units are in use, we cannot assume much about the final dimensions.
+        if (
+            (
+                $this->hasWidthUnit()
```

Comment:
It doesn't seem that `$this->hasWidthUnit()` is necessary since the regex will be run in `getWidthUnit`.

Reply:
Remove via https://github.com/ampproject/amp-toolbox-php/pull/49/commits/b66999c3816b115d408c69fb9451b655093cbda8